src/PluginClient.php

```diff
@@ -79,7 +78,7 @@ public function sendRequest(RequestInterface $request)
         $pluginChain = $this->createPluginChain($this->plugins, function (RequestInterface $request) {
             try {
                 return new FulfilledPromise($this->client->sendRequest($request));
-            } catch (HttplugException $exception) {
+            } catch (\Exception $exception) {
```

Comment:
> the Promise actually allows to receive a \Exception instance in the rejection callback, however Promises in the HTTP flow should only receive HTTP related exceptions, everything else should immediately be thrown.

That's why we may need to add an HttpPromise interface, and use this one, instead of the Promise, which enforce this and offer better visibility for devs when creating plugins and client.